src/sentry/snuba/discover.py

```diff
@@ -69,45 +69,57 @@ def is_real_column(col):
 
 
 def find_reference_event(reference_event):
-    try:
-        project_slug, event_id = reference_event.slug.split(":")
-    except ValueError:
-        raise InvalidSearchQuery("Invalid reference event")
+    columns = [field for field in reference_event.fields if is_real_column(field)]
+    return bulk_find_reference_event([reference_event], columns, "discover.find_reference_event")[0]
 
-    column_names = [resolve_column(col) for col in reference_event.fields if is_real_column(col)]
+
+def bulk_find_reference_event(reference_events, columns, referrer):
     # We don't need to run a query if there are no columns
-    if not column_names:
+    if not columns:
         return None
 
-    try:
-        project = Project.objects.get(
-            slug=project_slug,
-            organization=reference_event.organization,
-            status=ProjectStatus.VISIBLE,
-        )
-    except Project.DoesNotExist:
-        raise InvalidSearchQuery("Invalid reference event")
+    project_ids = set()
+    snuba_filter = eventstore.Filter(event_ids=[])
+    for reference_event in reference_events:
+        try:
+            project_slug, event_id = reference_event.slug.split(":")
+            project = Project.objects.get(
```

Comment:
I would pull this check out of the loop, and do one query with all the project IDs. If any of the projects don't exist, you can fail early.

Reply:
Is that a safe way to do it since organization is grabbed from reference_event?

Reply:
You could grab the org ID from the first reference event. They can't be from different orgs, right?

Reply:
Gonna be pulling this out from the PR, because we're no longer passing reference events from the frontend. But yeah the frontend doesn't have a way to do more than 1 org, but I still think it's potentially dangerous (while very edge case-y)